### PR TITLE
internal: update plugin example

### LINKED ISSUE #576: Update SP plugin example to use webpack
there is SP plugin with webpack and npm deps https://github.com/skyrim-multiplayer/skymp-research-client
it can be used for reference.

### PR DESCRIPTION
**Pull request by mrowrpurr**
closes #576 

`tl;dr` - Greatly simplified the `plugin-example` and added new functionality including:
- webpack
- zip for distribution
- walk-thru tutorial in README

---

View a preview of the `plugin-example` and its README here:
- https://github.com/mrowrpurr/skymp/tree/update-plugin-example/skyrim-platform/tools/plugin-example

---

Features:
- added very detailed README with walkthrough for new users
- webpack added to bundle dependencies: `npm run webpack` (_users will never have to run this themselves, just use `npm run dev`_)
- zip for distribution to mod sharing websites: `npm run zip`
- continuing to use `npm run dev` for development code watching
- moved to `npm` from `yarn` for ease of development for new users
- removed tests (_unused by most and more confusing than useful, we should revisit this in the future_)
- now configuration is done using a simple `skyrim.json` (and `package.json` for name) which contains most common path to Skyrim directory for most users (_the project should require zero configuration for most users_)
- optional `SKYRIMPATH` environment variable can be set instead of using `skyrim.json` which is useful for folks with many plugins
- removed tsc directory; now no custom scripts are present (_fewer files to reduce any confusion for for new users_); all deploy code is in `webpack.config.js`)
- no longer using templates such as tsconfig-default (_confusing for new users_)
- removed LICENSE (_users must choose their own license for their software_)
- no longer using TypeScript modules to user's local Skyrim Platform folder, using dependency on `@skyrim-platform/skyrim-platform` instead

Fixes #576 

Moved from previous @skyrim-platform/example-plugin repo (_which is now deleted so that all documentation lives in `skymp`_)

---

_First contributor notes: I didn't use **conventional commits** for all commit messages. My guess is that we 'Squash' the commit to a single commit with a conventional commit message? That is my plan :) I can re-push all commits rebased to a single commit with a conventional commit message is that is the process. Thanks!_

### CODE REVIEW

File: skyrim-platform/tools/plugin-example/README.md [hunk 1]
```diff
@@ -1,27 +1,250 @@
...
+    if (foldersInData)
+        Debug.messageBox(`Folders in Data directory: ${foldersInData.join("\n")}`)
+})
+```
+
+This is a simple demonstration of how to use one function from one common Skyrim library.
+
+For more documentation on how to use a third-party library, read the documentation on its npm page or GitHub repository.
+
+> Third-party library documentation is not provided on this website
```
Comment (Pospelove) [inline skyrim-platform/tools/plugin-example/README.md:207]: I'm not sure we should call it a website. Maybe wiki/readme/tutorial, etc.

File: skyrim-platform/tools/plugin-example/LICENSE.txt [hunk 2]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (Pospelove) [inline skyrim-platform/tools/plugin-example/LICENSE.txt:1]: SP is distributed under GPLv3 so as far as this change is ok, we should somehow point that plugin-example isn't polluted by GPL restrictions. We need to somehow say that they are able to do whatever they want with this bunch of configs and these three lines of code.

File: skyrim-platform/tools/plugin-example/skyrim.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "skyrimFolder": "C:/Program Files (x86)/Steam/steamapps/common/Skyrim Special Edition"
+}
```
Comment (Pospelove) [inline skyrim-platform/tools/plugin-example/skyrim.json:3]: Please also fix eof

File: skyrim-platform/tools/plugin-example/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    resolve: { extensions: ['.ts', '.tsx', '.js', '.jsx'] },
+    externals: {
+        // Do not bundle skyrim platform. Instead require('skyrimPlatform') at runtime
+        '@skyrim-platform/skyrim-platform': ['skyrimPlatform'], // <--- support import 'skyrimPlatform'
+        'skyrimPlatform': ['skyrimPlatform']
+    },
+    module: {
+        rules: [{ test: /\.tsx?$/, loader: 'ts-loader', options: { configFile: 'tsconfig.json' }}]
+    }
+};
```
Comment (Pospelove) [inline skyrim-platform/tools/plugin-example/webpack.config.js:78]: and here

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Comment by Pospelove** [comment]
Overall seems good! Fantastic documenting job.

File: skyrim-platform/tools/plugin-example/LICENSE.txt (hunk 2 above)
Comment (mrowrpurr) [reply skyrim-platform/tools/plugin-example/LICENSE.txt:1]: I added a bit about licensing under a `LICENSE` section at the bottom

> ### LICENSE
>
> Your plugin is your own software and can be licensed however you like!
> 
> Skyrim Platform is distributed under the GNU General Public License v3.0.

**Review by Pospelove — APPROVED** [review]

**Comment by Pospelove** [comment]
Thanks!

**Pull request merged**